tests/test_optuna_run_config_generator.py

```diff
@@ -0,0 +1,134 @@
+    OptunaRunConfigGenerator,
+)
+from model_analyzer.config.input.config_defaults import DEFAULT_BATCH_SIZES
+from model_analyzer.config.input.objects.config_model_profile_spec import (
+    ConfigModelProfileSpec,
+)
+from model_analyzer.device.gpu_device import GPUDevice
+from tests.common.test_utils import evaluate_mock_config
+
+from .common import test_result_collector as trc
+
+
+class TestOptunaRunConfigGenerator(trc.TestResultCollector):
+    def setUp(self):
+        self._default_max_batch_size = 4
+        self._test_config_dict = {
+            "name": "add_sub",
+            "input": [{"name": "INPUT__0", "dataType": "TYPE_FP32", "dims": [16]}],
+            "max_batch_size": self._default_max_batch_size,
+        }
+        with patch(
+            "model_analyzer.triton.model.model_config.ModelConfig.create_model_config_dict",
+            return_value=self._test_config_dict,
+        ):
+            self._mock_models = [
+                ModelProfileSpec(
+                    ConfigModelProfileSpec(model_name="add_sub"),
+                    MagicMock(),
+                    MagicMock(),
+                    MagicMock(),
+                )
+            ]
+
+        config = self._create_config()
+        self._rcg = OptunaRunConfigGenerator(
+            config=config,
+            gpus=[GPUDevice("TEST_DEVICE_NAME", 0, "TEST_BUS_ID0", "TEST_UUID0")],
+            models=self._mock_models,
+            client=MagicMock(),
+            model_variant_name_manager=ModelVariantNameManager(),
+            search_parameters=MagicMock(),
+            metrics_manager=MagicMock(),
+            seed=100,
+        )
+
+    def test_create_default_run_config(self):
+        """
+        Test that a default run config is properly created
+        """
+        default_run_config = self._rcg._create_default_run_config()
+
+        self.assertEqual(len(default_run_config.model_run_configs()), 1)
+        model_config = default_run_config.model_run_configs()[0].model_config()
+        perf_config = default_run_config.model_run_configs()[0].perf_config()
+
+        self.assertEqual(model_config.to_dict()["name"], self._test_config_dict["name"])
+        self.assertEqual(perf_config["batch-size"], DEFAULT_BATCH_SIZES)
+        self.assertEqual(
+            perf_config["concurrency-range"], 2 * self._default_max_batch_size
+        )
+
+    def test_create_objective_based_run_config(self):
+        """
+        Test that an objective based run config is properly created
+        """
+        trial = self._rcg._study.ask()
+        self._rcg._create_trial_objectives(trial)
+        run_config = self._rcg._create_objective_based_run_config()
+
+        model_config = run_config.model_run_configs()[0].model_config()
+        perf_config = run_config.model_run_configs()[0].perf_config()
+
+        self.assertEqual(model_config.to_dict()["name"], self._test_config_dict["name"])
+
+        # These values are the result of using a fixed seed of 100
```

Comment:
Are we at risk of this failing if optuna version changes? Looks like our requirements.txt uses > version instead of pinning an exact version.

Reply:
We might be. I don't know of a better way to lock this down.

Reply:
You can require exact version instead of >= in requirements.txt

Reply:
Done